File: app/src/main/java/com/aquamorph/frcmanager/decoration/Animations.java
package com.aquamorph.frcmanager.decoration;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.animation.Animation;
import android.view.animation.AnimationUtils;

import com.aquamorph.frcmanager.utils.Logging;

/**
 * Card slide in animations.
 *
 * @author Christian Colglazier
 * @version 1/7/2017
 */

public class Animations {

	public static void loadAnimation(final Context context, final View view,
	                                 final RecyclerView.Adapter adapter,
	                                 final Boolean firstLoad,
	                                 final Boolean isNewData) {
		if (context != null && view != null && adapter != null) {
			Animation slideOut = AnimationUtils.loadAnimation(context, android.R.anim
					.slide_out_right);
			final Animation slideIn = AnimationUtils.loadAnimation(context, android.R.anim
					.slide_in_left);
			slideOut.setAnimationListener(new Animation.AnimationListener() {
				@Override
				public void onAnimationStart(Animation animation) {}

				@Override
				public void onAnimationEnd(Animation animation) {
					Logging.INSTANCE.info(this, "Animation Ended", 1);
					(adapter).notifyDataSetChanged();
					view.startAnimation(slideIn);
				}

				@Override
				public void onAnimationRepeat(Animation animation) {}
			});
			if (isNewData) {
				if (firstLoad) view.startAnimation(slideIn);
				else view.startAnimation(slideOut);
			}
		}
	}
}

File: app/src/main/kotlin/com/aquamorph/frcmanager/decoration/Animations.kt
package com.aquamorph.frcmanager.decoration

import android.content.Context
import android.view.View
import android.view.animation.Animation
import android.view.animation.AnimationUtils
import androidx.recyclerview.widget.RecyclerView
import com.aquamorph.frcmanager.utils.Logging

/**
 * Card slide in animations.
 *
 * @author Christian Colglazier
 * @version 3/30/2018
 */

object Animations {

    /**
     * loadAnimation() manages animation for tab views.
     *
     * @param context app activity
     * @param view data view
     * @param adapter recyclerview adapter
     * @param firstLoad first time data has been seen
     * @param isNewData date has been updated
     */
    fun loadAnimation(
        context: Context?,
        view: View?,
        adapter: RecyclerView.Adapter<*>?,
        firstLoad: Boolean,
        isNewData: Boolean
    ) {
        if (context != null && view != null && adapter != null) {
            val slideOut = AnimationUtils.loadAnimation(context, android.R.anim.slide_out_right)
            val slideIn = AnimationUtils.loadAnimation(context, android.R.anim.slide_in_left)
            slideOut.setAnimationListener(object : Animation.AnimationListener {
                override fun onAnimationStart(animation: Animation) {}

                override fun onAnimationEnd(animation: Animation) {
                    Logging.info(this, "Animation Ended", 1)
                    adapter.notifyDataSetChanged()
                    view.startAnimation(slideIn)
                }

                override fun onAnimationRepeat(animation: Animation) {}
            })
            if (firstLoad) {
                view.startAnimation(slideIn)
            } else if (isNewData) {
                view.startAnimation(slideOut)
            }
        }
    }

    /**
     * loadMatchBreakdownAnimation() manages new activity animation.
     *
     * @param context activity context
     * @param view data view
     */
    fun loadMatchBreakdownAnimation(
        context: Context?,
        view: View?
    ) {
        if (context != null && view != null) {
            val slideIn = AnimationUtils.loadAnimation(context, android.R.anim.slide_in_left)
            view.startAnimation(slideIn)
        }
    }
}
